sources/packages/backend/apps/api/src/services/disbursement-receipt/disbursement-receipt.service.ts

```diff
@@ -18,12 +18,43 @@ export class DisbursementReceiptService extends RecordDataModelService<Disbursem
    * receipt belongs to.
    * @param studentId student to whom the disbursement
    * receipt belongs to.
+   * @param applicationId application id.
    * @returns disbursement receipt details.
    */
   async getDisbursementReceiptByAssessment(
     assessmentId: number,
     studentId?: number,
+    applicationId?: number,
   ): Promise<DisbursementReceipt[]> {
+    let findOptions: FindOptionsWhere<DisbursementReceipt> = {
```

Comment:
Can we try to use a similar approach from the suggested one here: https://github.com/bcgov/SIMS/pull/1984#discussion_r1213753966